File: js/sprint2/E/code.js
/*
Comment it before submitting
class Node {  
  constructor(value = null, next = null, prev = null) {  
    this.value = value;  
    this.next = next;  
    this.prev = prev;  
  }  
}
*/

function solution(node) {
    // Your code
    // ヽ(´▽`)/
}

function test() {
    var node3 = new Node("node3");
    var node2 = new Node("node2", node3);
    var node1 = new Node("node1", node2);
    var node0 = new Node("node0", node1);
    node1.prev = node0;
    node2.prev = node1;
    node3.prev = node2;
    var newHead = solution(node0);
    /*
    result is newHead === node3
    node0.prev === node1
    node1.next === node0
    node1.prev === node2
    node2.next === node1
    node2.prev === node3
    node3.next === node2
    */
}

File: go/sprint2/B/code.go
package main

// <template>
type ListNode struct {  
    data   string  
    next *ListNode  
}
// <template>

func Solution (head *ListNode) {
	// Your code
	// ヽ(´▽`)/
}

func test() {
	node3 := ListNode{"node3", nil}
	node2 := ListNode{"node2", &node3}
	node1 := ListNode{"node1", &node2}
	node0 := ListNode{"node0", &node1}
	Solution(&node0)
	/*
    Output is:
    node0
    node1
    node2
    node3
    */
}